src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Program.cs

```diff
@@ -125,23 +125,34 @@ static async Task RunAsync(ServerConfiguration serverConfiguration, Cancellation
 
     var languageServerLogger = loggerFactory.CreateLogger(nameof(LanguageServerHost));
 
-    var (clientPipeName, serverPipeName) = serverConfiguration.ServerPipeName is null
-        ? CreateNewPipeNames()
-        : (serverConfiguration.ServerPipeName, serverConfiguration.ServerPipeName);
+    LanguageServerHost? server = null;
+    if (serverConfiguration.UseStdIo)
+    {
+        server = new LanguageServerHost(Console.OpenStandardInput(), Console.OpenStandardOutput(), exportProvider, languageServerLogger, typeRefResolver);
+        // Redirect Console.Out to try prevent the standard output stream from being corrupted.
+        Console.SetOut(new StreamWriter(Console.OpenStandardError()));
```

Comment:
We should move this redirection to the top of RunAsync.

Reply:
On this point, should it be just after the logger is created?

```csharp
static async Task RunAsync(ServerConfiguration serverConfiguration, CancellationToken cancellationToken)
{
    // Create a console logger as a fallback to use before the LSP server starts.
    using var loggerFactory = LoggerFactory.Create(builder => ...);

    var logger = loggerFactory.CreateLogger<Program>();

    // Handle LSP communication channel method and Console.Out redirection
    if (serverConfiguration.UseStdIo)
    {
        if (serverConfiguration.ServerPipeName is not null)
        {
            logger.LogError("Server cannot be started with both --stdio and --pipe options.");
            Environment.Exit(1);
        }

        // Redirect Console.Out to try prevent the standard output stream from being corrupted.
        Console.SetOut(new StreamWriter(Console.OpenStandardError()));
    }
    
    ...
```